cli/src/semgrep/core_runner.py

```diff
@@ -871,6 +871,7 @@ def _run_rules_direct_to_semgrep_core(
 
             if ("time" in output_json) and core_output.time:
                 self._add_match_times(profiling_data, core_output.time)
+                profiling_data.set_max_memory_bytes(core_output.time.max_memory_bytes)
```

Comment:
For consistency I would rather also have this set by a `self._add_max_memory_bytes()` function